cmd/rpcdaemon/commands/eth_receipts.go

```diff
@@ -275,15 +275,32 @@ func (api *APIImpl) GetTransactionReceipt(ctx context.Context, txnHash common.Ha
 	var ok bool
 
 	blockNum, ok, err = api.txnLookup(ctx, tx, txnHash)
-	if !ok {
-		// It is not an ideal solution (ideal solution requires extending TxnLookupReply proto type to include bool flag indicating absense of result),
-		// but 0 block number is used here to mean that the transaction is not found
-		return nil, nil
+	if err != nil {
+		return nil, err
 	}
+
+	cc, err := api.chainConfig(tx)
 	if err != nil {
 		return nil, err
 	}
 
+	if !ok && cc.Bor == nil {
+		return nil, nil
+	}
+
+	// if not ok and cc.Bor != nil then we might have a bor transaction
+	if !ok {
+		blockNumPtr, err := rawdb.ReadBorTxLookupEntry(tx, txnHash)
```

Comment:
do you think read kv.BorTxLookup must be "after" (not "before") txnLookup?

Reply:
Well it is both the same in my opinion.